src/extension/extension.ts

```diff
@@ -37,13 +35,20 @@ import { BazelBuildCodeLensProvider } from "../codelens";
 import { BazelTargetSymbolProvider } from "../symbols";
 import { BazelWorkspaceTreeProvider } from "../workspace-tree";
 
+/**
+ * Allows logging to the custom output channel "VSCode-Bazel".
+ */
+export let logOutputChannel: vscode.OutputChannel | undefined;
```

Comment:
This would require other components to import `extension/extension` in order to access this channel, which seems like the wrong dependency direction (and could lead to cycles). `extension/extension` is essentially the "main" file that shouldn't be imported by anyone else.

Instead, an approach like this might work: create a new `src/logging/logging.ts` file (along with the `index.ts` following the pattern of the other directories) and move the channel variable and setup function in there. Then, also create and export a log method (`bazelLog`?) that logs to the channel declared in that file. This lets you keep the channel itself private and only export the setup and logging functions, which clients would call.

Reply:
Done.